p2p/protocol/identify/id_glass_test.go

```diff
@@ -36,10 +37,11 @@ func TestFastDisconnect(t *testing.T) {
 
 		// Kill the connection, and make sure we're completely disconnected.
 		s.Conn().Close()
```

Comment:
The problem here was that _sometimes_, we'd end up with two connections (for example, one TCP and one QUIC connection). Of course, calling `s.Conn().Close()` here would only close one of them, which is why the loop on the next line would never terminate.

Reply:
Alternative fix would be to loop over all connections and close them.

Reply:
actually, let's do that -- it's the right thing to do.

Reply:
ok, done